Source/Core/Core/FileMonitor.h

```diff
@@ -0,0 +1,24 @@
+// Copyright 2008 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <memory>
+
+#include "Common/CommonTypes.h"
+
+namespace DiscIO
+{
+class IFileSystem;
+class IVolume;
+}
+
+namespace FileMonitor
+{
+// Can be called with nullptr to set the file system to nothing. When not called
+// with nullptr, the volume must remain valid until the next SetFileSystem call.
+void SetFileSystem(const DiscIO::IVolume* volume);
+// SetFileSystem must have been called before calling Log
```

Comment:
Oh, you mean asserting it in `Log`? We can't do that. The requirement is that `SetFileSystem` has been called at some point and that the passed-in volume is still valid (unless it was `nullptr`). When `Log` is called more than once after each `SetFileSystem` call (which happens all the time), `s_filesystem` will not be `nullptr` – this is completely normal behavior.

Actually, the current comment might be a little bit misleading. It's not that calling `Log` without `SetFileSystem` is forbidden, it's just that it won't be able to log anything, like if someone would have called `SetFileSystem` with `nullptr` manually. I'll change it.